src/frontend/screens/Library/components/GamesList/index.tsx

```diff
@@ -26,8 +56,66 @@ const GamesList = ({
   onlyInstalled = false,
   isRecent = false
 }: Props): JSX.Element => {
-  const { gameUpdates } = useContext(ContextProvider)
+  const { gameUpdates, showNonAvailable } = useContext(ContextProvider)
   const { t } = useTranslation()
+  const [gameCards, setGameCards] = useState<JSX.Element[]>([])
+
+  useEffect(() => {
+    const createGameCards = async () => {
+      if (!library.length) {
+        return
+      }
+      const resolvedLibrary = library.map(async (gameInfo) => {
+        const {
+          app_name,
+          is_installed,
+          runner,
+          install: { is_dlc }
+        } = gameInfo
+        if (is_dlc) {
+          return null
+        }
+        if (!is_installed && onlyInstalled) {
+          return null
+        }
+
+        let isAvailable = is_installed
+
+        if (is_installed) {
+          isAvailable = await handleNonAvailableGames(app_name, runner)
+          console.log('isAvailable', isAvailable)
+        }
+
+        const hasUpdate = is_installed && gameUpdates?.includes(app_name)
+        return (
+          <GameCard
+            key={app_name}
+            hasUpdate={hasUpdate}
+            buttonClick={() => handleGameCardClick(app_name, runner, gameInfo)}
+            isAvailable={isAvailable}
+            forceCard={layout === 'grid'}
+            isRecent={isRecent}
+            gameInfo={gameInfo}
+          />
+        )
+      })
+      const gameCardElements = (await Promise.all(
+        resolvedLibrary
+      )) as JSX.Element[]
+
+      setGameCards(gameCardElements)
+    }
+
+    createGameCards()
+  }, [
+    library,
+    onlyInstalled,
+    layout,
+    gameUpdates,
+    isRecent,
+    nonAvailbleGamesArray,
+    showNonAvailable
+  ])
```

Comment:
I think `useMemo` is a better hook to use here, this is kinda recreating what useMemo does but it will trigger an extra render when calling setGameCards

Reply:
good point, fixed it